sycl/include/CL/sycl/detail/scheduler/commands.hpp

```diff
@@ -38,8 +38,8 @@ enum BlockingT { NON_BLOCKING = 0, BLOCKING };
 
 // The struct represents the result of command enqueueing
 struct EnqueueResultT {
-  enum ResultT { SUCCESS, BLOCKED, FAILED };
-  EnqueueResultT(ResultT Result = SUCCESS, Command *Cmd = nullptr,
+  enum ResultT { SYCL_SUCCESS, SYCL_BLOCKED, SYCL_FAILED };
```

Comment:
I suggest to convert this to [scoped enumeration](https://en.cppreference.com/w/cpp/language/enum) (`enum class`) instead. Someone could define `SYCL_SUCCESS` as well as just `SUCCESS` and the issue will still be there

Reply:
That won't work because the pre-processor will change all occurrences of SUCCESS into 0.
The pre-processor macro expansion has no knowledge of scoped enumeration or whatever.
It just scans all texts and replaces the match with its definition which is "0".
So, it is better to use a name that won't be used by the user. "SYCL_" prefix is quite unlikely used by the user.